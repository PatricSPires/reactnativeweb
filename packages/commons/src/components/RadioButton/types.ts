import React from 'react'
import { ThemeType } from '../../styles/types'

export type RadioButtonProps = {
  onPress?: () => void
  children?: React.ReactNode
  checked?: boolean
  disabled: boolean
  theme?: ThemeType
}
