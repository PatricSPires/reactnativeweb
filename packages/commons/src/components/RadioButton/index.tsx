import React from 'react'
import { StyleSheet, View } from 'react-native'
import { RadioButtonProps } from './types'

import {
  Container,
  RadioButtonWrapper,
  RadioButtonCheck,
  RadioButtonText,
} from './styles'

const RadioButton = ({
  children,
  checked,
  onPress,
  disabled,
}: RadioButtonProps) => (
  <Container
    accessibilityRole="radio"
    disabled={disabled}
    checked={checked}
    onPress={onPress}
  >
    <View style={styles.wrapper}>
      <RadioButtonWrapper checked={checked} disabled={disabled}>
        <RadioButtonCheck checked={checked} disabled={disabled} />
      </RadioButtonWrapper>
      {children && (
        <RadioButtonText disabled={disabled}>{children}</RadioButtonText>
      )}
    </View>
  </Container>
)

RadioButton.defaultProps = {
  disabled: false,
  checked: false,
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export default RadioButton
