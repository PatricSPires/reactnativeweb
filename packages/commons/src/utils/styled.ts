import styleSheetSerializer from 'jest-styled-components/src/styleSheetSerializer'

export function enableStyledComponentsSerializer() {
  expect.addSnapshotSerializer(styleSheetSerializer)
}
