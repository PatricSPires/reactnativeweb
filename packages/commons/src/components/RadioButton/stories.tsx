import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import RadioButton from './index'
import { RadioButtonProps } from './types'

export default {
  title: 'Formulário/Radio Button',
  component: RadioButton,
  parameters: {
    docs: {
      description: {
        component:
          'Este é o nosso componente de radio button, que possui alguns estilos com base nas propriedades e estados.',
      },
    },
  },
  args: {
    children: 'Place your text here',
    checked: false,
    disabled: false,
  },
  argTypes: {
    onPress: {
      action: 'clicked',
      description: 'Muda o estado do componente para verdadeiro ou falso.',
    },
    children: {
      description: 'Texto que acompanha o radio button. Não é obrigatório.',
    },
    checked: {
      description: 'Alterna o estado do componente entre verdadeiro e falso.',
    },
    disabled: {
      description:
        'Altera o estado do componente para desabilitado e não permite clique.',
    },
  },
}

export const Basic: Story<RadioButtonProps> = (args) => (
  <RadioButton {...args} />
)

export const Active: Story<RadioButtonProps> = Basic.bind({})
Active.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __checked__:true.',
    },
  },
}

Active.args = {
  checked: true,
}

export const Disabled: Story<RadioButtonProps> = Basic.bind({})
Disabled.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __disabled__:true.',
    },
  },
}

Disabled.args = {
  disabled: true,
  checked: false,
}

export const ActiveDisabled: Story<RadioButtonProps> = Basic.bind({})
ActiveDisabled.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __disabled__:true e __checked__:true.',
    },
  },
}

ActiveDisabled.args = {
  disabled: true,
  checked: true,
}
