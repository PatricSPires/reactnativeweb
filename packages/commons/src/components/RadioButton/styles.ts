import styled from 'styled-components'
import { TouchableWithoutFeedback, View, Text } from 'react-native'
import theme from '../../styles/theme'
import { RadioButtonProps } from './types'

export const RadioButtonWrapper = styled(View)`
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  ${({ checked, disabled }: RadioButtonProps) => {
    if (checked === false && disabled) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
      `
    }
    return ``
  }}
`

export const RadioButtonCheck = styled(View)`
  width: 8px;
  height: 8px;
  border-radius: 100%;

  ${({ checked, disabled }: RadioButtonProps) => {
    if (checked && checked === true && !disabled) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked && checked === true && disabled && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const RadioButtonText = styled(Text)`
  margin-left: 8px;
  color: ${(props: RadioButtonProps) =>
    props.disabled && props.disabled === true
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: ${(props: RadioButtonProps) =>
    props.disabled && props.disabled === true ? 'not-allowed' : 'pointer'};

  &:focus ${RadioButtonWrapper} {
    border-color: ${theme.supportColors.information};
    box-shadow: 0px 0px 0px 1px ${theme.supportColors.information};
    ${({ checked, disabled }: RadioButtonProps) => {
      if (checked === true && disabled === false) {
        return `
          box-shadow: 0px 0px 0px 1px ${theme.supportColors.information};
        `
      }
      return ``
    }}
  }
`
