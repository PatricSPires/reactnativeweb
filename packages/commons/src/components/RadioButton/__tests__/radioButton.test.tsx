import * as React from 'react'
import { render } from '@testing-library/react-native'
import RadioButton from '../index'
import { enableStyledComponentsSerializer } from '../../../utils/styled'

beforeAll(() => {
  enableStyledComponentsSerializer()
})

describe('Radio Button Test', () => {
  it('should render the radio button', () => {
    const { toJSON } = render(<RadioButton />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the radio button disabled true', () => {
    const { toJSON } = render(<RadioButton disabled />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the radio button disabled false', () => {
    const { toJSON } = render(<RadioButton disabled={false} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the radio button disabled checked true', () => {
    const { toJSON } = render(<RadioButton checked />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the radio button disabled checked false', () => {
    const { toJSON } = render(<RadioButton checked={false} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the radio button with children', () => {
    const { toJSON } = render(<RadioButton>Hello world</RadioButton>)

    expect(toJSON()).toMatchSnapshot()
  })
})
