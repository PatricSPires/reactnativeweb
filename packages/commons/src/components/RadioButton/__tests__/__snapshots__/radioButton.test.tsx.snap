// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Test should render the radio button 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
</div>
`;

exports[`Radio Button Test should render the radio button disabled checked false 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c4:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c5:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
</div>
`;

exports[`Radio Button Test should render the radio button disabled checked true 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
  background-color: #161616;
}

.c4:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c5:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
</div>
`;

exports[`Radio Button Test should render the radio button disabled false 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c4:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
</div>
`;

exports[`Radio Button Test should render the radio button disabled true 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-color: #e0e0e0;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
}

.c4:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  aria-disabled={true}
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
>
  <div
    aria-disabled={true}
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={true}
  >
    <div
      aria-disabled={true}
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={true}
    />
  </div>
</div>
`;

exports[`Radio Button Test should render the radio button with children 1`] = `
.c2 {
  width: 18px;
  height: 18px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  width: 8px;
  height: 8px;
  border-radius: 100%;
}

.c4 {
  margin-left: 8px;
  color: #161616;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c5:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c6:focus .c1 {
  border-color: #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="radio"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
  <div
    className="c4 sc-dlnjPT c4 css-text-901oao"
    dir="auto"
    disabled={false}
  >
    Hello world
  </div>
</div>
`;
